doc/r1-design.md

```diff
@@ -0,0 +1,148 @@
+
+
+## Preliminary policy set: Supporting free5gc SMF's dependency on free5gc UPF
+Different vendors' NFs may have different requirements that are tied with the expected NF deployment topology. For example, free5gc SMF configuration requires explicit config on the list of UPFs that are connected to it, with explicit config on the IP address bind to both of each's N3 and N4 interfaces.
+
+A general solution should be creating a policy interface for NTC --- where for R1, we only support one policy: NFs with NFType "X" should not be deployed until all its connected NFs with NFType "Y" are deployed. In this case, "X" is SMF and Y is UPF. NTC will render this request as follows:
+
+- for all NFInstances with NFType SMF:
+    - identify all the connected UPFs via NFTopology
+    - as each UPF related PR is created, NTC would add conditions wait-for-UPF-xxx:false to corresponding package
+    - set these aforementioned conditions to **true** only when the corresponding UPF PR reaches PUBLISHED state
+
+# Alternative Solution
+It is also possible that specifically for the free5gc case, the responsibility for handling this case can fall squarely on the IPAM controller. IPAM controller today tracks allocated IP address for each request / purpose; if both free5gc SMF and UPF packages have IP address allocation condition for that UPF's N3 and N4 interfaces, then regardless of which package triggers the initial IP allocation request, the request would yield the same IP address, and therefore would serve this
+particular purpose.
+
+## Tracking Packages
+NTC will embed a label 'nf-deployment-name', which is set to NFToplogy CR's own name; PackageVariantSet and PackageVariant controllers will propagate this label to all the PackageRevision resources associated with the "fan-out"ed packages.
+
+NTC watches over all PackageRevision resources in the management cluster, and maps the NFTopology intent to the number of deployed NF resources via tracking corresponding PackageRevision resources. As each PackageRevision resource gets to *PUBLISHED* state, NTC would update NFDeployedTopology resource to reflect on newly target NF deployment to monitor.
+
+NTC will also extract the number of pending conditions from each of the PR as a display to user(s) who want to continuously examine the lifecycle of the packages, and which conditions are gating a particular package from being deployed. NTC essentially tracks status of package deployment for all NF instances specified derived from the NFTopology intent; however, NTC does **NOT** track the progress of the deployment once the package is pushed to the workload cluster, those statues
```

Comment:
@MuhammadSaheerCheruvath @ganchandrasekaran This could be something we look at for the Nephio tab in the UI